app/components/new-offer-link.js

```diff
@@ -2,12 +2,12 @@ import Ember from 'ember';
 const { getOwner } = Ember;
 
 export default Ember.Component.extend({
-
+  store: Ember.inject.service('store'),
   messageBox: Ember.inject.service(),
   i18n: Ember.inject.service(),
 
   allOffers: Ember.computed(function(){
-    var store = this.get('targetObject.store');
+    var store = this.get('store');
     return store.peekAll("offer");
```

Comment:
can directly write this.get('store').peekAll("offer")... No need to declare extra variable